app/firebase/functions/src/posts/hooks/post.updated.hook.ts

```diff
@@ -1,65 +1,129 @@
 import {
+  ActivityEventBase,
+  ActivityEventCreate,
+  ActivityType,
+  PostActData,
+} from '../../@shared/types/types.activity';
+import {
+  AppPost,
   AppPostParsedStatus,
   AppPostRepublishedStatus,
 } from '../../@shared/types/types.posts';
-import { AutopostOption, PLATFORM } from '../../@shared/types/types.user';
 import { logger } from '../../instances/logger';
 import { createServices } from '../../instances/services';
 import { enqueueTask } from '../../tasks.support';
+import { UsersHelper } from '../../users/users.helper';
 import { AUTOPOST_POST_TASK } from '../tasks/posts.autopost.task';
 import { PARSE_POST_TASK } from '../tasks/posts.parse.task';
 
-export const postUpdatedHook = async (postId: string) => {
-  const { db, time, users, postsManager } = createServices();
+const PREFIX = 'POST-UPDATED-HOOK';
+
+// TODO: change interface to receive post as the after value and also send the previous one
+export const postUpdatedHook = async (post: AppPost, postBefore?: AppPost) => {
+  const postId = post.id;
+
+  const { db, time, activity, users } = createServices();
 
+  /** Frontend watcher to react to changes in real-time */
   const updateRef = db.collections.updates.doc(`post-${postId}`);
   const now = time.now();
 
-  logger.debug(`postUpdatedHook post-${postId}-${now}`);
+  logger.debug(`postUpdatedHook post-${postId}-${now}`, undefined, PREFIX);
 
   await db.run(async (manager) => {
     manager.set(updateRef, { value: now });
   });
 
-  /** check if it should be auto-parsed or auto published */
-  const { post, author } = await db.run(async (manager) => {
-    const post = await postsManager.processing.posts.get(postId, manager, true);
-    const author = await users.repo.getUser(post.authorId, manager, true);
-    return { post, author };
+  /** Handle post create */
+  if (postBefore === undefined) {
+    // trigger parsePostTask
+    logger.debug(`triggerTask ${PARSE_POST_TASK}-${postId}`);
+    await enqueueTask(PARSE_POST_TASK, { postId });
+  }
+
+  const activitiesCreated: ActivityEventBase[] = [];
+
+  /** Create the activity elements */
+  const { wasParsed } = await db.run(async (manager) => {
+    /** detect parsed state change */
+    const wasParsed =
+      postBefore &&
+      postBefore.parsedStatus === AppPostParsedStatus.UNPROCESSED &&
+      post.parsedStatus === AppPostParsedStatus.PROCESSED;
+
+    if (wasParsed) {
+      logger.debug(`wasParsed ${PARSE_POST_TASK}-${postId}`, undefined, PREFIX);
+      const event: ActivityEventCreate<PostActData> = {
+        type: ActivityType.PostParsed,
+        data: {
+          postId: post.id,
+        },
+        timestamp: time.now(),
+      };
+
+      const parsedActivity = activity.repo.create(event, manager);
```

Comment:
It looks like this creates the PostParsed activity for every user, which will lead to the issue above (if I'm understanding correctly)